gems/sorbet-runtime/test/types/props/serializable.rb

```diff
@@ -1001,6 +1001,10 @@ class RedundantEnumStruct < T::Struct
     prop :deprecated_enum, T.all(MyEnum, T.deprecated_enum(MyEnum.values))
   end
 
+  class UnknownValueStruct < T::Struct
+    prop :unknownable, T.any(MyEnum, T::Props::UnknownValue)
+  end
+
```

Comment:
Can we add a _lot_ more tests?

Some tests:

- using `UnknownValue` with some structs that fails to deserialize (wrong types, missing args)
- using `UnknownValue` with a `BSON` document, where the serialized value is bytes not a string
- using `UnknownValue` with an enum that serializes to a `Symbol` instead of a `String`

Reply:
Also maybe worthwhile to test `T.nilable(T.any(MyEnumType, UnknownValue))` and/or `T.any(T.nilable(MyEnumType), UnknownValue))`.